test/11_array.jl

```diff
@@ -101,6 +101,7 @@ for ordered in (false, true)
         @test levels(x2) == levels(x)
 
         x2 = copy(x)
+        copy!(x2, x)
```

Comment:
You're hijacking an existing test, so `copy` isn't tested at all now. Please rather add copies of this test block at the end of both `11_array.jl` and `12_nullablearray.jl`. I think we need to test several different cases as these can be tricky:
- levels in a different order than index: use an unsorted input like `["Old", "Middle", "Young"]` as source and `[ "Young", "Old", "Middle"]` as destination to test that.
- differing levels between source and destination.
- for nullable categorical array: test when some missing values are present, and when only missing values are present.